File: Jarvis_free/backend/modules/speech_text.py
"""
Текст для озвучки: блоки между разделителями --- только для чтения на экране.
"""

from __future__ import annotations

import re

_DETAIL_SPLIT = re.compile(r"(?:^|\n)\s*---+\s*(?:\n|$)", re.MULTILINE)


def strip_detail_blocks(text: str) -> str:
    """
    Оставляет фрагменты вне пар --- … --- (чётные части после split).
    Пример: «Кратко» + --- + «Подробности» + ---  →  озвучивается только «Кратко».
    """
    if not text or not text.strip():
        return ""
    parts = _DETAIL_SPLIT.split(text.strip())
    if len(parts) == 1:
        return _clean_for_speech(parts[0])
    spoken_parts = [parts[i].strip() for i in range(0, len(parts), 2) if parts[i].strip()]
    return _clean_for_speech("\n".join(spoken_parts))


def _clean_for_speech(text: str) -> str:
    t = text.strip()
    t = re.sub(r"\*\*([^*]+)\*\*", r"\1", t)
    t = re.sub(r"`([^`]+)`", r"\1", t)
    t = re.sub(r"\[([^\]]+)\]\([^)]+\)", r"\1", t)
    t = re.sub(r"#{1,6}\s*", "", t)
    t = re.sub(r"[🔊🔑⚠️❌✅📎🎙️📁📊🖼️⚙️💬🗑️✏️📥📱🟠]", "", t)
    t = re.sub(r"\s+", " ", t).strip()
    return t


def chat_message_with_details(short: str, detail: str | None = None) -> str:
    """Короткая часть в чате + подробности между --- (не озвучиваются)."""
    short = (short or "").strip()
    detail = (detail or "").strip()
    if not detail:
        return short
    return f"{short}\n\n---\n{detail}\n---"

File: Jarvis_free/frontend/src/lib/speechText.ts
/**
 * Озвучка: между --- … --- только текст для экрана (не читается вслух).
 */

const DETAIL_SPLIT = /(?:^|\n)\s*---+\s*(?:\n|$)/gm

export function stripDetailBlocksForSpeech(text: string): string {
  const raw = text.trim()
  if (!raw) return ''
  const parts = raw.split(DETAIL_SPLIT).map((p) => p.trim())
  if (parts.length === 1) return cleanForSpeech(parts[0])
  const spoken = parts.filter((_, i) => i % 2 === 0).filter(Boolean)
  return cleanForSpeech(spoken.join('\n'))
}

function cleanForSpeech(text: string): string {
  return text
    .replace(/\*\*([^*]+)\*\*/g, '$1')
    .replace(/`([^`]+)`/g, '$1')
    .replace(/\[([^\]]+)\]\([^)]+\)/g, '$1')
    .replace(/#{1,6}\s*/g, '')
    .replace(/[🔊🔑⚠️❌✅📎🎙️📁📊🖼️⚙️💬🗑️✏️📥📱🟠]/gu, '')
    .replace(/\s+/g, ' ')
    .trim()
}

/** Короткая строка для озвучки + подробности между --- для экрана. */
export function formatChatMessage(spoken: string, detail?: string): string {
  const s = spoken.trim()
  const d = detail?.trim()
  if (!d) return s
  return `${s}\n\n---\n${d}\n---`
}
